Upgrade to Rails 5 and Rspec 3.5

valadzko opened the pull request:
Hi, we have upgraded the gem to support Rspec 3.5 and Rails 5
I don't think it will be compatible with previous version, so maybe we will need to update gem version, too.

moretonb commented:
Hi @valadzko, @Hirurg103, @AliakseiLeanovich,

Thank you for the contribution! We'll take a look at it. I think you are right and we will need to go up a major version in the gem and version.rb to 1.0.0 too.

moretonb commented on spec/account_spec.rb:
```diff
@@ -31,180 +31,209 @@
+
+    subject { account.message_status(message_id) }
+
+    it "should post to the message dispatcher resource" do
```
Can you rename this test to match the case. Something like "should get the messageheader resource with a message id"

valadzko commented:
@moretonb travis failed first time because of the using Ruby version < 2.2.2. 
I updated .travis ruby version to 2.2.3 (it is now minimum required version), and also I added 2.3.0 and 2.4.0 versions.

moretonb commented:
I've tagged current master and included information to indicate this PR is a breaking change for existing consumers. Between that and a major version, I'm happy that should indicate strongly enough that we are removing support for older Ruby versions.

Thanks again guys, we'll get this accepted very soon.

valadzko commented:
Hi, thank you for the review and the good news that it will be merged!
From my point of view there are some more things that could be done here before (or early after) the merge:
1) Change min version of Rails to "4.1.0" instead of current "5.0.2". It will cover more customers, actually I think to have min Rails 5 is a very strict requirement for now.
2) Change one config file to three different, each for production/test/dev environment. I did not manage to finish it properly and now we can see the warning message when run rspec tests.
3) Create a proper documentation: we need at least to say that Rails 3 is not supported anymore and to add an info about Rails 5.
4) Investigate if we need a dummy app inside of the gem(it seems like it is only for rails 3 projects) and remove it if it's indeed only for rails 3.

@moretonb , if you have any plans to merge and upgrade the current gem(seems like you are), could you please let me know if I can take care on some of the items from this list?

SLStinson commented:
Hi @valadzko! Thanks very much for the contribution. We would like to merge this, and we are happy for you to look into doing the work for points 1, 2 and 4 from your list to be included in the pull request. We will take care of the work for the documentation as there are other aspects of it we would like to improve.
